### PR TITLE
More unit tests for events

### PR DESCRIPTION
**Pull request by CelticMinstrel**
This just exists to ensure CI passes, but if anyone wants to comment on the new tests, feel free.

### CODE REVIEW

File: data/test/macros/wml_unit_test_macros.cfg [hunk 1]
```diff
@@ -88,6 +88,7 @@ Elvish Archer#endarg
                 name = "Alice"
                 type = {SIDE1_LEADER}
                 id=alice
+                random_traits=no
```
Comment (stevecotton) [inline data/test/macros/wml_unit_test_macros.cfg:91]: Why is this line necessary? Alice and Bob already don't get random traits, so if this line is needed then it indicates an API change that's subtle but which will affect UMC.

Comment (CelticMinstrel) [reply data/test/macros/wml_unit_test_macros.cfg:91]: They don't get random traits? How do you know? I can remove it if it's redundant.

Comment (stevecotton) [reply data/test/macros/wml_unit_test_macros.cfg:91]: I ran a test within the GUI (ran it with `--test`) and checked.

Comment (CelticMinstrel) [reply data/test/macros/wml_unit_test_macros.cfg:91]: Why do they not get random traits if they don't have this line…?

**Comment by stevecotton** [comment]
> Why do they not get random traits if they don't have this line…?

I'm not sure, but think it's something about only generating deterministic numbers during scenario creation, thus no random numbers for the traits. Or it might be because they're leaders.

> Note: I don't see how this change would affect UMC though.

If someone has balanced their scenario assuming that the enemy leader doesn't have the `quick` trait then it might break. I can remember an example with a leader capturing a village, and then on turn 2 moving to a tiny castle that was closer than their main base.

One of the things that the unit tests do is to ensure that things don't change during a stable branch. If someone finds the reason that Alice and Bob don't get traits and 'fixes' it then, without this line, the unit tests will notice that Alice randomly has more MP.

**Comment by CelticMinstrel** [comment]
Oh, come to think of it, it's probably because they're leaders. Okay, I'll remove that line then.

**Comment by CelticMinstrel** [comment]
The CI test failures aren't in the WML unit tests, so I'm merging this.

**Comment by stevecotton** [comment]
> The CI test failures aren't in the WML unit tests, so I'm merging this.

Yes they are, they're your bug. After your opposition in #5616 to making the CI logs easier to read, it's disappointing to see that you didn't read them before merging.

```
20210409 14:27:32 error validation: Invalid key 'accuracy=' in tag [attack]
at test/scenarios/events-test_autovariables.cfg:252
    included from test/_main.cfg:29
    included from data/_main.cfg:68
20210409 14:27:32 error validation: Extra tag [recruit]; there may only be 1 [recruit] in [do_command]
at core/macros/utils.cfg:152
    included from core/_main.cfg:19
    included from data/_main.cfg:18
20210409 14:27:32 error validation: Invalid key 'accuracy=' in tag [attack]
at test/scenarios/events-test_filters.cfg:156
    included from test/_main.cfg:29
    included from data/_main.cfg:68
validation failed
Test validation complete!  Success: No
```
